File: Data Structures & Algorithms/is-anagram/submission-8.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        counts = {}
        for ch in s:
            counts[ch] = counts.get(ch, 0) + 1
        for ch in t:
            if ch not in counts:
                return False
            counts[ch] -= 1
            if counts[ch] == 0:
                counts.pop(ch)
        
        return not counts;

File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> counts;
        for (int num : nums) {
            counts[num] += 1;
        }

        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> min_heap;

        for (auto& pair : counts) {
            min_heap.push({pair.second, pair.first});
            if (min_heap.size() > k) {
                min_heap.pop();
            }
        }

        vector<int> res;

        while (k > 0) {
            res.push_back(min_heap.top().second);
            min_heap.pop();
            --k;
        }

        return res;
    }
};
